arangod/Aql/Functions.cpp

```diff
@@ -2518,31 +2513,31 @@ AqlValue Functions::DateFromParameters(
 
       // All Parameters have to be a number or a string
       if (!value.isNumber() && !value.isString()) {
-        RegisterInvalidArgumentWarning(query, funcName.c_str());
+        RegisterInvalidArgumentWarning(query, AFN);
         return AqlValue(AqlValueHintNull());
       }
     }
 
     // Parse the Year
     years y{ExtractFunctionParameterValue(parameters, 0).toInt64(trx)};
```

Comment:
Could you `||` the checks so just one if-body needs to be implemented?

Reply:
I don't get it. its a loop over all params, they have to be string or number. what could be changed here? This part of the patch barely switches the function name from the outside to `const char *`

Reply:
```
years y{ExtractFunctionParameterValue(parameters, 0).toInt64(trx)};
months m{ExtractFunctionParameterValue(parameters, 1).toInt64(trx)};
days d{ExtractFunctionParameterValue(parameters, 2).toInt64(trx)};

if ( y .. || .. || .. days{0} ) {
 // same code here
}
```